### PR TITLE
DRILL-7094: UnSupported Bson type: DECIMAL128 Mongodb 3.6 onwards

### PR DESCRIPTION
**Pull request by rajeshguptaiiitm**
DRILL-7094: UnSupported Bson type: DECIMAL128 Mongodb 3.6 onwards

# [DRILL-7094](https://issues.apache.org/jira/browse/DRILL-7094): UnSupported Bson type: DECIMAL128 Mongodb 3.6 onwards

## Description

Handling the Decimal128 by converting into Double.

## Documentation
Now Bson Decimal128 type will be handled for the SQL queries and Show the result in double value.

## Testing
(Created a Document in Mongo DB with field of Decimal128 type and executing a SQL query using Apache Drill and it return the results. Previously there was a error unsupported Bson Type.)

### CODE REVIEW

**Comment by rajeshguptaiiitm** [comment]
Created the new pull request to merge

**Review by cgivre — APPROVED** [review]
LGTM +1

**Comment by cgivre** [comment]
@rajeshguptaiiitm 
Thanks for your persistence and I'm sorry we couldn't figure out git.  (I've been there before.... ) The commit looks good to me, and I've approved.  I'll merge tomorrow morning unless there are any objections.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/bson/BsonRecordReader.java
Comment (vvysotskyi) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/bson/BsonRecordReader.java:183]: Drill has VarDecimal data type, please use it instead of double.

Comment (rajeshguptaiiitm) [reply exec/java-exec/src/main/java/org/apache/drill/exec/store/bson/BsonRecordReader.java]: Resovle it

**Comment by rajeshguptaiiitm** [comment]
Done the changes as per review comment.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/bson/BsonRecordReader.java
Comment (vvysotskyi) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/bson/BsonRecordReader.java:375]: Writers have `writeVarDecimal()` method that handles storying `BigDecimal` values. Please use that method:
```suggestion
    if (isList) {
      writer.list.varDecimal().writeVarDecimal(readBigDecimal);
    } else {
      writer.varDecimal(fieldName, readBigDecimal.precision(), readBigDecimal.scale()).writeVarDecimal(readBigDecimal);
    }
```

Comment (rajeshguptaiiitm) [reply exec/java-exec/src/main/java/org/apache/drill/exec/store/bson/BsonRecordReader.java]: Done the suggested changes.

Comment (vvysotskyi) [reply exec/java-exec/src/main/java/org/apache/drill/exec/store/bson/BsonRecordReader.java]: You haven't removed the code where creating varDecimalHolder, it is not required for now.

Comment (rajeshguptaiiitm) [reply exec/java-exec/src/main/java/org/apache/drill/exec/store/bson/BsonRecordReader.java]: Sorry, that was a miss from side. Now I have done it.

**Review by vvysotskyi — APPROVED** [review]
@rajeshguptaiiitm, thanks for making changes, +1

**Comment by rajeshguptaiiitm** [comment]
Could you please help me to merge those changes because I don't have the rights.

**Comment by vvysotskyi** [comment]
@rajeshguptaiiitm, Github CI job failed, please fix it before we will proceed with the merge

**Comment by cgivre** [comment]
@rajeshguptaiiitm 
Thanks again for the contribution. Before we merge, we need to squash the commits into one commit.  Are you clear on the process to do that and what we need?

**Comment by rajeshguptaiiitm** [comment]
Could you please share the command again to do that ??

**Comment by cgivre** [comment]
> Could you please share the command again to do that ?? As you can see Í have 5 commits now.

Sure.  Here's a link:
https://gist.github.com/patik/b8a9dc5cd356f9f6f980#the-harder-and-less-flexible-way

The procedure I follow is `The Harder and Less Flexible Way`.  In your case, I believe the specific command are:

```
git reset --soft HEAD~5
git commit -m "DRILL-7094: UnSupported Bson type: DECIMAL128 Mongodb 3.6 onwards"
git push origin +master
```

Don't feel bad... I botched this my first few times. ;-)

**Comment by cgivre** [comment]
@rajeshguptaiiitm 
One more thing.. In the future, if you plan on making additional contributions to Drill, I'd suggest creating new branches for the feature rather than working from your `master` branch.  It's not a requirement, but it will make your life easier.

**Comment by rajeshguptaiiitm** [comment]
Thanks for you great help and really appreciate.
In future I will remember your suggestion to push changes. If you want you can assign an issue to me related to Mongo DB because mainly I am working on Mongo DB.
Now please merge my changes.

**Comment by cgivre** [comment]
@rajeshguptaiiitm 
Thanks for the contribution.  Your changes are merged!  If there are other improvements you could suggest to the Mongo plugin, we welcome the contributions!

**Pull request merged**
